### PR TITLE
[ptq] Support PPL computation

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit supports PPL computation.

Related: https://github.com/Samsung/TICO/issues/89
Draft: https://github.com/Samsung/TICO/pull/212
TICO-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

```bash
python tico/experimental/quantization/ptq/examples/compare_ppl.py
Calibrating with 50,000 tokens.

Creating UINT-8 clone …
`rope_scaling`'s original_max_position_embeddings field must be less than max_position_embeddings, got 2048 and max_position_embeddings=2048
Calibrating UINT-8 observers …
Using the latest cached version of the dataset since wikitext couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'wikitext-2-raw-v1' at /home/sw4670.chae/.cache/huggingface/datasets/wikitext/wikitext-2-raw-v1/0.0.0/b08601e04326c79dfdd32d625aee71d232d685c3 (last modified on Fri May  3 21:06:41 2024).
Calibration: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 22/22 [00:03<00:00,  6.20it/s]

Calculating perplexities …
Using the latest cached version of the dataset since wikitext couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'wikitext-2-raw-v1' at /home/sw4670.chae/.cache/huggingface/datasets/wikitext/wikitext-2-raw-v1/0.0.0/b08601e04326c79dfdd32d625aee71d232d685c3 (last modified on Fri May  3 21:06:41 2024).
Token indices sequence length is longer than the specified maximum sequence length for this model (289077 > 131072). Running this sequence through the model will result in indexing errors
PPL:  99%|██████████████████████████████████████████████████████████████████████████████████████████████████▎| 561/565 [07:52<00:03,  1.19it/s]

┌── Wikitext-2 test perplexity ─────────────
│ UINT-8 :  9969.43
└───────────────────────────────────────────
```

### CODE REVIEW

File: tico/experimental/quantization/ptq/utils/metrics.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        target_ids = input_ids.clone()
+        target_ids[:, :-trg_len] = -100  # mask previously-scored tokens
+
+        with torch.no_grad():
+            outputs = model(input_ids, labels=target_ids)
+            # loss is already averaged over non-masked labels
+            neg_log_likelihood = outputs.loss
+
+        # exact number of labels that contributed to loss
+        loss_tokens = (target_ids[:, 1:] != -100).sum().item()
```
Comment (dayo09) [inline tico/experimental/quantization/ptq/utils/metrics.py:103]: How about exposing an api to set this `ignore_index` from outer side, and name it as `ignore_index`?
I see that your `DummyLM` carefully set the value as `ignore_index` for CrossEntropyLoss function. However some model can have different `ignore_index` value.

In the same context, meta llama may use another ignore_index, so it's safe to get one from tokenizer's `pad_token_id`.
So I suggest to

1. expose ignore_index for perplexity function.
2. set pad_token_id as ignore_index in example meta's case

Comment (mhs4670go) [reply tico/experimental/quantization/ptq/utils/metrics.py:103]: Good catch! 

I think exposing `ignore_index` at the API level makes sense. Different models (or loss setups) may not always use -100, so allowing it to be passed in makes the function more general and future-proof.

One thing to be careful about: using `pad_token_id` as `ignore_index` is not generally correct for Hugging Face causal LMs (including Meta LLaMA). By convention, they always use -100 as the masked-label value for `CrossEntropyLoss`, while `pad_token_id` is still a valid token ID and should not be ignored. So this is what I'll do.

1. Expose `ignore_index` as a parameter (default -100).
2. Only use `pad_token_id` as `ignore_index` if the specific model is designed that way. For HF causal LMs, keep -100.
3. If `ignore_index=None`, auto-detect from `model.config.ignore_index` when available, otherwise fall back to -100.

This way the API is flexible for non-standard models, but stays consistent with the standard HF behavior. Thanks for the review.

**Review by dayo09 — APPROVED** [review]
LGTM Thanks!

**Review by seockho-kim — APPROVED** [review]
LGTM
